### DISCUSSION TITLE
USR-TCP232-E2: Welche Version funktioniert

### DISCUSSION THREAD

**Discussion by Nesselbrand67** [Tom-Bom-badil/samson_trovis_557x#49] [General (de/en)]
Hallo zusammen,

ich bin gerade ein wenig verwirrt, welcher Adapter von PUSR der Richtige ist. Meine FW Station mit einer Trovis  5573 wird erst im Dezember geliefert, so dass ich zur Zeit nur Trockenübungen machen kann.
Als Vorbereitung wollte ich schonmal den ModbusTCP Adapter zusammenbasteln und konfigurieren.

Bei Ebay und Ali finde ich jeweils zwei Versionen des E2, eine mit STM chip mit M7 Core und eine von TI mit M4 Core.
Die Webseiten von PUSR sind hier nicht so eindeutig, unter [https://www.pusr.com/products/dual-ttl-to-ethernet-modules-usr-tcp232-e2.html](url) ist ein Mischmasch von M7 und M4 zu finden, unter [https://www.pusr.com/products/dual-UART-to-ethernet-module-usr-tcp232-e2.html](url) wird vom M7 geredet mit einem vergleich zu der alten M4 Version.

Mit scheint, dass Tom die M7 Version in Betrieb hat, das Bild zeigt ja auch deutlich den STM Chip.
Auch beim K2 gab es wohl ein Update vom M4 zum M7 ( [https://www.pusr.com/products/ethernet-to-uart-modules-usr-k7.html](url)).

Das Problem ist nun, das der Reseller mir mitgeteilt hat, dass die M7 Version abgekündigt ist und er nur noch die alte M4 Version versenden kann (Ebay Seite:  [https://www.ebay.de/itm/256529102877](url))

Was nun? Ich stehe da gerade ein wenig ratlos da. 
Könnte der alte M4 noch funktionieren? 
Habt ihr Reseller, die gesichert die M7 Version verschicken? Bei Ali und Ebay steht manchmal M7, aber Bild vom TI Chip, oder umgekehrt, hier würde ich auf nichts vertrauen...

**Comment by Tom-Bom-badil** [comment]
Die 'Hardware darunter' hat mich zugegebenerweise nie wirklich interessiert - meine Kriterien waren: Läuft / läuft nicht. Über irgendwelche Inkompatibilitäten der M4 ./ .M7-Basis zum Projekt ist aber bisher nichts berichtet worden- somit sollten auch beide laufen.

Da hilft vermutlich nur ausprobieren. Die M4-Version ist ja sogar genügsamer hinsichtlich des Stromverbrauchs. Ich sehe keinen Grund, warum die nicht funktionieren sollte ...

/tom

_p.s. Falls von Interesse: Es wird voraussichtlich demnächst der ESP32 zu den Bauanleitungen dazukommen (und damit endlich die oft angefragte WLAN-Anbindung). Einige haben diesen ja schon erfolgreich über ESPHome im Einsatz. Ich plane jedoch eine andere Firmware, so dass ESPHome als Drittkomponente nicht mehr notwendig sein wird. Die USR-Adapter sollen ohne viel Aufwand durch den ESP ersetzt werden können; und das für jede Software-Anbindung, die heute schon läuft. Dauert aber noch ein bisschen - erst will ich die Lovelace-Seiten für HA fix und fertig haben, bevor ich neue Baustellen aufmache._

**Reply by Nesselbrand67** [reply]
Interessant, könnte der ESP32 auch TCP/IP? Ich bin noch Old-School und lass meine kritischen Systeme nicht über WLAN laufen...

**Reply by Tom-Bom-badil** [reply]
Klar. TCP/IP ist ja nur das Protokoll - und dem ist egal, ob es über Funk oder Kabel geht (Tech-Sprech: Die Kommunikation läuft dann halt über andere Technik auf Layer 1 (Übertragungsschicht) und 2 (Sicherungsschicht)).

Falls Du mit TCP/IP eine verkabelte Verbindung meinst: Ja, es gibt auch ESP's mit Netzwerkanschluss; meines Wissens sogar welche mit POE, wo die Versorgungsspannung über das LAN-Kabel kommt..

Ich werde übrigens bei 'verkabelt' bleiben - der K7 ist einfach so schön kompakt, praktisch und stabil, und gleich neben der Übergabestation ist eine Netzwerkdose. Das ESP-Projekt ist für die, bei denen eine Verkabelung nicht oder nur mit großem Aufwand möglich ist, und die deshalb eine WLAN-Anbindung brauchen.

/tom

**Reply by Nesselbrand67** [reply]
Huch, da war ich unpräzise in meiner Frage. Ja, ich wollte wissen, ob es auch ESP32 Versionen mit integriertem 10/100 PHY gibt. 
Diese scheint es ja zu geben (sagt zumindest Tante Google) und daher wäre meine nächste Frage als ESP32-Neuling, ob die Firmware dann alle möglichen PHYs unterstützen wird, oder ob ich für die 10/100 PHY Unterstützung noch Module einbinden muss.

Ich werde mal abwarten und schaun, was geht wenn meine FW Station da ist. 

Der K7 ist mir zu klein, die Löterei machen meine Augen nicht mehr mit...

**Comment by Nesselbrand67** [comment]
Kurzes Update zu den E2 Versionen (Antwort des USR Supports auf meine Anfrage):

> Currently our new version is USR-TCP232-E2-H7, M4 version has been discontinued.
> But M4 version also supports Modbus RTU to Modbus TCP function, it can be configured via webpage, the default IP is same with H7 version: 192.168.0.7. 

Scheint, dass der Reseller, den ich erwischt hatte, seine Restbestände loswerden will...

**Reply by Tom-Bom-badil** [reply]
Ja, den Eindruck hatte ich nach Deiner Beschreibung gestern auch. Es hat irgendwie keinen Sinn ergeben, dass das neuere Modell (H7/M7) zwar auf der [Herstellerseite](https://www.pusr.com/products/dual-UART-to-ethernet-module-usr-tcp232-e2.html) stark beworben, aber angeblich *discontinued* ist, während das als 'old' bezeichnete Modell (M4) das angeblich aktuelle sei. Macht irgendwie keinen Sinn.

/tom
